<!doctype html>

<head>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta charset="utf-8">
    <!-- Bootstrap v4beta Imports -->
    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0-beta/css/bootstrap.min.css" integrity="sha384-/Y6pD6FV/Vv2HJnA6t+vslU6fwYXjCFtcEpHbNJ0lyAFsXTsjBbfaDjzALeQsN6M" crossorigin="anonymous">

    <style media="screen">
        body {
            padding-top: 70px;
            padding-bottom: 70px;
        }
    </style>

    <!-- Syntax Highlighter. Use both pygments and hl.js. -->
    <link rel="stylesheet" href="../../../static/pygments.css">
    <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/default.min.css">
    <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
    <script>hljs.initHighlightingOnLoad();</script>

    <!-- Google Analytics -->
    <script>
        (function(i, s, o, g, r, a, m) {
            i['GoogleAnalyticsObject'] = r;
            i[r] = i[r] || function() {
                (i[r].q = i[r].q || []).push(arguments)
            }, i[r].l = 1 * new Date();
            a = s.createElement(o),
                m = s.getElementsByTagName(o)[0];
            a.async = 1;
            a.src = g;
            m.parentNode.insertBefore(a, m)
        })(window, document, 'script', 'https://www.google-analytics.com/analytics.js', 'ga');

        ga('create', 'UA-12498603-2', 'auto');
        ga('send', 'pageview');
    </script>

    <!-- ClustrMaps Tracking -->
    <!-- <script type="text/javascript" id="clstr_globe" src="//cdn.clustrmaps.com/globe.js?d=nhKoDpoTjWz4pC6CwI-fSy4hPoJ1uXwTLCfMCT3OK_8"></script> -->

    <!-- FontAwesome embed -->
    <script src="https://use.fontawesome.com/cb9dbe8e41.js"></script>

    <!-- Custom CSS -->
    <link rel="stylesheet" href="/static/css/custom.css">
</head>

<title>Blog - Eric J. Ma's Personal Site</title>
<body class="body">
    <nav class="navbar navbar-expand-sm navbar-light fixed-top bg-light">
        <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#local-links" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
            <span class="navbar-toggler-icon"></span>
        </button>
        <div class="collapse navbar-collapse" id="local-links">
            <ul class="navbar-nav mr-auto">
                
                <li class="nav-item">
                    <a class="nav-link" href="/"><i class="fa fa-home" aria-hidden="true"></i> Home</a>
                </li>
                
                <li class="nav-item">
                    <a class="nav-link" href="/resume"><i class="fa fa-file-text" aria-hidden="true"></i> Resume</a>
                </li>
                
                <li class="nav-item">
                    <a class="nav-link" href="/blog"><i class="fa fa-rss" aria-hidden="true"></i> Blog</a>
                </li>
                
                <li class="nav-item">
                    <a class="nav-link" href="/open-source"><i class="fa fa-code" aria-hidden="true"></i> Open Source</a>
                </li>
                
                <li class="nav-item">
                    <a class="nav-link" href="/projects"><i class="fa fa-briefcase" aria-hidden="true"></i> Projects</a>
                </li>
                
                <li class="nav-item">
                    <a class="nav-link" href="/talks"><i class="fa fa-microphone" aria-hidden="true"></i> Talks</a>
                </li>
                
                <li class="nav-item">
                    <a class="nav-link" href="/teaching"><i class="fa fa-university" aria-hidden="true"></i> Teaching</a>
                </li>
                
            </ul>
        </div>
    </nav>
    <div class="container">
        
  
    
  <!-- <div class="blog-post"> -->
  <link rel="stylesheet" href="../../../static/pygments.css">
  <div>
  
    <h1><a href="../../../blog/2016/6/16/baarn-2016-bullet-point-notes/">BAARN 2016 Bullet Point Notes</a></h1>
  
  <p class="meta">
    written by
    
      <a href="https://twitter.com/ericmjl">Eric J. Ma</a>
    
    on 2016-06-16
  </p>
  <p>I attended the 2016 Boston Area Antibiotic Resistance Network meeting, held at the American Association for the Advancement of Science (AAAS). Here’s my bullet-point notes version of what went on.</p>
<p><strong>Session 1: New antibiotic strategies and therapeutics:</strong></p>
<p>In this section, scientists and group leaders presented novel ways of thinking about anti-bacterial treatment.</p>
<ol>
<li>Small molecules: focus on how to get molecules into the cell. (Anita Miller, Entasis)</li>
<li>Phage lysins: one component of phages, add externally to a bacteria. (Vincent Fischetti, Rockefeller University)</li>
<li>Ecobiotics: Consortia of commensal bacteria to restore general diversity, or impact specific pathways. (Matthew Henn, Seres Therapeutics)</li>
<li>Chemical discovery: Gene clusters can predict biochemical activity of natural products, great for discovering new chemicals. Cool point I took home - a gene cluster that encodes a molecule that destroys function of a common protein may also encode a resistant version of that version. (Emily Balskus, Harvard University)</li>
</ol>
<p><strong>Session 2: Creating a sustainable ecosystem for antibiotic production.</strong></p>
<p>In this section, representatives from finance, manufacturing, regulatory, and public policy discussed how they thought a sustainable ecosystem for producing antibiotics could be fostered.</p>
<p>Panel members:</p>
<ul>
<li>Joseph Larsen, BARDA</li>
<li>Cara Cassino, Contrafect</li>
<li>Edward Cox, FDA</li>
<li>Alan Carr, Needham &amp; Co.</li>
</ul>
<p>Are there enough ideas in the pipeline?</p>
<ul>
<li>BARDA model: product accelerator; public-private partnerships.</li>
<li>Missing: a “pull” incentive - if I develop X successfully, I will get paid Y. Need lawmakers’ help.</li>
</ul>
<p>Are there enough ideas in the pipeline? </p>
<ul>
<li>BARDA model: product accelerator; public-private partnerships.</li>
<li>Missing: a “pull” incentive - if I develop X successfully, I will get paid Y. Need lawmakers’ help.</li>
<li>There’s a move towards targeted drug therapies.</li>
<li>New paradigms come with unknown risks.</li>
<li>Market around antibiotics: there’s underlying interest, but some skepticism and fear towards biotech.</li>
<li>There are sub-sections of antibiotics space (e.g. gram positives) that can do well, because of unmet need.</li>
<li>Regulatory: yet, if there’s an unmet need, then we’re behind. Private vs. societal value are not aligned.</li>
<li>Antibiotics: a bit like fire department. We don’t pay fire department by the number of times they put out fires, but as an insurance. Payments are de-linked from the amount of usage.</li>
<li>Private sector model isn’t going to foster investment.</li>
<li>Incentives need to factor in not just pathogens, but also clinical manifestations of the pathogens.</li>
</ul>
<p>Diagnostics?</p>
<ul>
<li>Rapid, within 1-2 hours, which can be good enough to form the basis for decision making, is an unmet need.</li>
<li>Susceptibility testing is also important.</li>
</ul>
<p>Why is there not reduced antibiotic usage in agriculture?</p>
<ul>
<li>Companies that manufacture these products are told to voluntarily remove indications for non-therapeutic uses.</li>
<li>By end of year, antibiotics will need to be prescribed by a vet.</li>
<li>Antibiotics have growth promotion and prophylactic non-therapeutic uses. </li>
<li>Interestingly: high prices per course discourages lax usage. Interesting…… But what if the economic &amp; regulatory model was changed?</li>
</ul>
<p>New antibiotics are a bit like the different types of fire extinguishers (one for petrochemical, one for electrical, etc.). You only buy a new fire extinguisher if it is sufficiently different. New antibiotics have to likewise be sufficiently different.</p>
<p>What’s the one thing that has to be done?</p>
<ul>
<li>From small antibiotic developer’s perspective: prioritization is needed. Favourable reimbursement is needed. Also support the fire department model, but don’t discriminate between large and small developers.</li>
<li>Push incentives are helping, but we need the “pull” incentives, i.e. the known return on investment.</li>
<li>Parallels between rare diseases and antibiotics. "Fire department”-style incentives might be a good way forward.</li>
<li>Rapid, high resolution diagnostics would be really transformative, for example, in their use in clinical trials for rare infections.</li>
<li>Clinical trial networks, infrastructure to be able to study new, narrow-spectrum agents.</li>
</ul>
<p><strong>Session 3: Entrepreneurship</strong></p>
<p>Things to note:</p>
<ul>
<li>Make sure the science is solid. Nature cannot be fooled.<ul>
<li>Regulatory issues must be thought of right from the beginning. </li>
</ul>
</li>
<li>Technology can come from anywhere. A good company is one that addresses an unmet need.</li>
<li>Talk with regulatory people early on. They sometimes have productive things to say as well.</li>
<li>Get a great team together. They should know the market, believe in the mission.</li>
<li>Focus on proof of concept and differentiation. If these are achieved, good things can happen.</li>
<li>Talk with people you trust, who are willing to tell you “this idea is a bad idea”.</li>
</ul>
<p><strong>Session 4: Diagnostics</strong></p>
<ul>
<li>From talking with others over lunch, I found that diagnostics have to be technically so simple (i.e. solid phase chemistry). </li>
<li>Clinical goals:<ul>
<li>Reduce use of antibiotics</li>
<li>Rapid diagnosis and treatment</li>
<li>Data to guide the most appropriate antibiotic</li>
<li>Clinicians need to know when to withhold and when to deploy antibiotics.</li>
</ul>
</li>
<li>Unmet clinical needs:<ul>
<li>Rapid, simple, inexpensive to distinguish viral from bacterial infection, or if a person is even infected or not.<ul>
<li>CLIA-waived - so simple, a non-laboratory person can do it.</li>
<li>Likely a biomarker test.</li>
</ul>
</li>
<li>Rapid, simple inexpensive test to detect viral pathogens<ul>
<li>There are simple molecular assays, but none that are inexpensive.</li>
</ul>
</li>
<li>Rapid identification of bacteria<ul>
<li>Laboratory-based test</li>
<li>Improvements are being made.</li>
</ul>
</li>
<li>Rapid susceptibility testing<ul>
<li>Still a challenge.</li>
</ul>
</li>
</ul>
</li>
<li>Rapid: a context-specific term.<ul>
<li>Bacterial vs. viral: 15-20 minutes.</li>
<li>Viral detection: 15-20 minutes, on-demand and not batched.</li>
<li>Bacterial detection: an hour or less, on-demand.</li>
<li>Susceptibility: \&lt; 6 hours.</li>
</ul>
</li>
<li>Identification of species is better than gram pos/neg.</li>
<li>T2 Biosystems: an interesting platform, PCR-based detection. </li>
<li>Resistance testing: <ul>
<li>Genotypic is helpful for gram-positive strains. Gram-negatives are more challenging.</li>
<li>Phenotypic testing: Accelerate Diagnostics and Geneweave. Accelerate uses FISH and growth characteristics. </li>
</ul>
</li>
<li>How to change a clinician’s practice?<ul>
<li>Reliable test.</li>
<li>Outcomes data - prove to doctor that test makes a difference.</li>
<li>Reimbursement model that encourages use of test.</li>
<li>Educate physicians - what and when to order, how to interpret results. (Interpretation is key!)</li>
</ul>
</li>
<li>T2 diagnostics: magnetic resonance-based detection of pathogens. </li>
<li>What is the best economic model for infectious disease products?</li>
</ul>

  </div>

    <p><i>Did you enjoy this blog post? <a href="../../../blog/2016/6/16/baarn-2016-bullet-point-notes/#disqus">Let's discuss more</a>!</i></p>
    <hr class="fancy">
  
    
  <!-- <div class="blog-post"> -->
  <link rel="stylesheet" href="../../../static/pygments.css">
  <div>
  
    <h1><a href="../../../blog/2016/6/16/boston-area-antibiotic-resistance-network-baarn-meeting-2016/">Boston Area Antibiotic Resistance Network (BAARN) Meeting 2016</a></h1>
  
  <p class="meta">
    written by
    
      <a href="https://twitter.com/ericmjl">Eric J. Ma</a>
    
    on 2016-06-16
  </p>
  <p>Yesterday I attended a very fruitful <a href="https://www.eventbrite.com/e/baarn-2016-boston-area-antibiotic-resistance-network-tickets-22500283955">BAARN 2016 meeting</a>. This was the fourth annual meeting, and the range of topics spanned the scientific (new technologies for treating and diagnosing) to economic (incentives for development of new antibiotics and diagnostic technologies) to medical (outlining the real unmet needs that doctors are facing). I learned a ton; the $100 registration fee was definitely worthwhile.</p>
<p>There were three things I took note of that I think could be translated over from the bacterial surveillance world to the viral surveillance world.</p>
<p><strong>(1) Genomic and Phenotypic Surveillance</strong></p>
<p>This one was a big take-home for me, given the research direction I’m hoping to pursue after grad school.</p>
<p>Genomic surveillance is the use of genome sequence to determine the evolutionary history and potential risk of a pathogen. In the bacterial world, genomic surveillance efforts are mostly concentrated on finding known antibiotic resistance genes.</p>
<p>Phenotypic surveillance involves actually experimentally testing a pathogen for some phenotype. In the bacterial world, this means testing a cultured isolate against a panel of drugs.</p>
<p>In my mind, these two are inseparable. Genomic surveillance will always be cheaper and faster to carry out, given the advances in portable sequencers; phenotypic surveillance is going to be rate-limited by biochemistry. On the other hand, only systematic phenotypic measurements can give us the necessary data to link genome sequence to phenotype. In my view, the function that connects genome/protein sequence to phenotype is complex enough that for pure surveillance (not scientific) purposes, machine learning models are probably the best tools to use. With tools like <a href="http://www.randalolson.com/2016/05/08/tpot-a-python-tool-for-automating-data-science/">TPOT</a>, it should be trivial to automate the selection of best predictive features and models. The tough part, then, is generating a gold-standard, epidemiologically relevant dataset.</p>
<p><strong>(2) Incentives</strong></p>
<p>Another interesting point that came out is the notion that antibiotic drugs are the only drug whose value depreciates with its widespread usage. The main factor at play here is the emergence of drug resistant bacteria. This makes it financially disadvantageous for companies to come in and invest in making new drugs.</p>
<p>One alternative incentive system that was brought up is best described with an analogy - fire stations. Taxpayers pay into a common pool of money, which funds fire stations and firemen. As opposed to paying firemen per fire that they put out, we pay them a fixed amount as a public good/insurance against fire disasters. Likewise, as opposed to paying drug companies per dosage of antibiotics that are sold, a better incentive may be to guarantee payment at (peer-/externally-reviewed) drug development milestones, in addition to a sum for maintaining an arsenal of drugs that could be stewarded when needed.</p>
<p><strong>(3) Rapid Detection</strong></p>
<p><a href="https://vivo.brown.edu/display/ac184">Dr. Angela Caliendo</a> (Brown University) brought up some really, really useful points during her talk, which I took note of in my <a href="../../../blog/2016/6/16/baarn-2016-bullet-point-notes/">bullet-point notes</a>. Rapid detection was the point that stuck the most.</p>
<p>“Rapid” is a nebulous concept, so Angela’s talk helped bring some clarity here. For the physician (and by my own extension, front-line epidemiologist), the major unmet medical needs for countering infectious disease outbreaks are:</p>
<ol>
<li>Distinguishing bacterial from viral infection.</li>
<li>Identifying bacterial pathogen.</li>
<li>Identifying viral pathogen</li>
<li>Measuring susceptibility of pathogen.</li>
</ol>
<p>In each case, “rapid” means different things. To distinguish bacterial from viral infection, and to be most useful in guiding physician decision-making, “rapid” means within minutes, at the bedside, doable by a non-laboratory trained person. For viral detection, it has to likewise be within 15-20 minutes to be “rapid”. For bacterial detection, an hour or less is best, and for susceptibility testing, less than 6 hours is optimal.</p>
<p>Knowing these points really helps set the engineering constraints properly. Basically, any useful assay has to work within minutes, be operable by a non-laboratory trained person, and accept blood/swabs as input. Solid phase (bio)chemistry is going to be really important.</p>
<p><strong>Synthesis/Thoughts</strong></p>
<p>My current projects are geared towards tying viral genotype to phenotype, with the goal of predicting a virus’ pathogenic risk profile from its sequence. Because sequence data is the input, and not blood/swab samples, I won’t be playing in the rapid detection space. However, I think the tools I hope to develop will play well in the susceptibility prediction/determination space.</p>
<p>With the MinION sequencer, the sequence of a virus can be determined within one day of isolation. Now, the sequencer’s <a href="https://www.engadget.com/2014/09/18/minion-usb-dna-sequencer-beta-test/">60-85% accuracy</a> (this number comes from a 2014 popular news report, and <a href="http://www.nature.com/nbt/journal/v33/n3/full/nbt.3103.html">a recent 2015 paper</a> supports this number) isn’t up to the point where single point mutations can be identified accurately, so I think it’s best used for viral species/subtype identification (e.g. in the case of influenza). When the accuracy moves below 1 error in 10,000 base calls (and I’m confident it will given the efforts of scientists and engineers to make this happen), that’s when I think that the susceptibility prediction tools I’m trying to develop will come in handy; common viral genomes sizes are on the order of thousands of bases. The key (and hard part) here is investing in <em>good measurement data</em> on which models can be trained that map sequence to phenotype.</p>

  </div>

    <p><i>Did you enjoy this blog post? <a href="../../../blog/2016/6/16/boston-area-antibiotic-resistance-network-baarn-meeting-2016/#disqus">Let's discuss more</a>!</i></p>
    <hr class="fancy">
  
    
  <!-- <div class="blog-post"> -->
  <link rel="stylesheet" href="../../../static/pygments.css">
  <div>
  
    <h1><a href="../../../blog/2016/6/15/gaining-focus-going-analog/">Gaining Focus: Going Analog</a></h1>
  
  <p class="meta">
    written by
    
      <a href="https://twitter.com/ericmjl">Eric J. Ma</a>
    
    on 2016-06-15
  </p>
  <p>I have found that I most successfully focus when I have my scribble book by my side. Jotting down what I'm going to do next on the notebook focuses my mind more than jotting it down in Evernote, which is done digitally. I think it's the combination of having it recorded on a companion "screen" to my MacBook Air.</p>

  </div>

    <p><i>Did you enjoy this blog post? <a href="../../../blog/2016/6/15/gaining-focus-going-analog/#disqus">Let's discuss more</a>!</i></p>
    <hr class="fancy">
  
    
  <!-- <div class="blog-post"> -->
  <link rel="stylesheet" href="../../../static/pygments.css">
  <div>
  
    <h1><a href="../../../blog/2016/6/14/how-to-give-a-great-data-science-workshop/">How to give a great Data Science Workshop</a></h1>
  
  <p class="meta">
    written by
    
      <a href="https://twitter.com/ericmjl">Eric J. Ma</a>
    
    on 2016-06-14
  </p>
  <p>I recently was asked by <a href="http://www.hugobowneanderson.com">Dr. Hugo Bowne-Anderson</a> (a really chill guy, who probably would insist that I drop the "Dr.") to contribute some of my thoughts on what makes a great data science workshop. Having been on both sides as an instructor and a learner, here's some of my thoughts.</p>
<p><strong>What distinguishes a Data Science Workshop from a Data Science talk or lecture?</strong></p>
<p>I think the distinguishing mark of a data science workshop is that the instructor will make sure that there is hands-on coding involved. The only exception to this that I have experienced was in <a href="http://clearscience.info/wp/">Bang Wong's</a> workshop on data visualization, where we did group discussion instead. In both cases, participant involvement is very important.</p>
<p><strong>What are the 3 most important qualities for a great Data Science Workshop to have and the 3 most important for it to NOT have? Why (one sentence on each quality)?</strong></p>
<p>My top 3 important qualities are:</p>
<ol>
<li><strong>Interactive &amp; hands-on:</strong> in line with the MIT motto <a href="https://libraries.mit.edu/mithistory/institute/seal-of-the-massachusetts-institute-of-technology/"><em>Mens et Manus</em></a>, the best workshops apply theory soon after learning it.</li>
<li><strong>Targeted:</strong> The best workshops have their difficulty tailored to their audience’s skill level.</li>
<li><strong>Expandable knowledge:</strong> Ideally, the workshop isn’t the end of learning; the best workshops point students to other resources to further the learning.</li>
</ol>
<p>It should be evident that the opposites of the above should be avoided, but I’ll put out a few ideas of common good teaching practices that might (paradoxically) be best avoided in a workshop setting:</p>
<ol>
<li><strong>Repetition:</strong> I think there should be as little conceptual repetition as possible; while it’s good for learning, but I think it’s not suited for a workshop setting, where I think the goal is a “sufficiently deep introductory survey” (an oxymoron of terms).</li>
<li><strong>Lecturing:</strong> This is admittedly tricky; there has to be enough lecturing, but not too much, and I think the best instructors intersperse them sparingly throughout the workshop.</li>
<li><strong>Handling questions:</strong> Questions are great during the breaks and after, but I’ve seen that the best instructors, and most cooperative students as well, defer their tougher and more detailed questions till a later time.</li>
</ol>
<p><strong>How do you avoid losing your audience in the 1st 5 minutes?</strong></p>
<p>The most important thing I learned was to keep the energy high. My verbal and body language have to show that I'm excited to share my knowledge, thrilled to meet everybody present, and eager to continue the conversation afterwards. The best instructors that I’ve seen, such as <a href="http://www.allendowney.com">Allen Downey</a>, do this. Following that, I think it’s important to set expectations, to clarify what will and won’t be covered.</p>
<p><strong>What tools/software/resources do you use when giving a Workshop? What tools/software do you ask your attendees to use?</strong></p>
<p>The topic I teach the most has been <a href="https://github.com/ericmjl/Network-Analysis-Made-Simple">network/graph analysis fundamentals</a>, using the <a href="https://networkx.github.io">NetworkX</a> API to introduce these ideas. For this, I use <a href="https://www.python.org">Python</a> (and the <a href="http://scipy.org">scientific Python stack</a>, in particular), the <a href="http://jupyter.org">Jupyter notebook</a>, and <code>git</code> on <a href="https://github.com">GitHub</a>. I like using the <a href="https://github.com/damianavila/RISE">RISE plugin</a> for Jupyter, which instantly turns my code and markdown cells into slides that I can advance through. Everybody complains about font sizes being too small; RISE solves that instantly for the notebook, and I can spice up the theme by changing the CSS.</p>
<p>Apart from that, where possible, I try to make use of sticky notes. It’s something I picked up from my <a href="http://software-carpentry.org">Software</a> &amp; <a href="http://www.datacarpentry.org">Data Carpentry</a> instructor training. Green stands for "I'm good to go!", while red stands for "I'm in need of some help."</p>
<p><strong>What would you like to find in a “How to teach a great Data Science Workshop” article?</strong></p>
<p>I’d like to see real feedback from students about their instructors. Nothing beats having direct feedback from the participants. I’d also like to see a list of questions that instructors should ask themselves (or have a coach-like figure ask them) after each workshop, so that we can train ourselves to focus on the most important aspects of teaching.</p>

  </div>

    <p><i>Did you enjoy this blog post? <a href="../../../blog/2016/6/14/how-to-give-a-great-data-science-workshop/#disqus">Let's discuss more</a>!</i></p>
    <hr class="fancy">
  
    
  <!-- <div class="blog-post"> -->
  <link rel="stylesheet" href="../../../static/pygments.css">
  <div>
  
    <h1><a href="../../../blog/2016/6/13/form-and-function/">Form and Function</a></h1>
  
  <p class="meta">
    written by
    
      <a href="https://twitter.com/ericmjl">Eric J. Ma</a>
    
    on 2016-06-13
  </p>
  <p>When writing code, it’s often a great idea to write tests in parallel. Today I experienced how important it is to have a split window. I write code on my left pane, and I can immediately set up the corresponding test on the right pane. Just as such.</p>
<p><img src="../../../blog/2016/6/13/form-and-function/split-screens.png" alt="split screens"></p>
<p>I used to not write my tests immediately, and I’m right now suffering that technical debt in parts of my code. That was before learning that Sublime Text 3 allows me to do split window panes (with the shortcut key <code>Cmd+1/2/3/4/5</code>). Now that I know it, writing tests in parallel with my code is much easier. Form indeed informs function.</p>

  </div>

    <p><i>Did you enjoy this blog post? <a href="../../../blog/2016/6/13/form-and-function/#disqus">Let's discuss more</a>!</i></p>
    <hr class="fancy">
  

  
  <div class="pagination">
    
      <a href="../../../blog/page/22/">&laquo; Previous</a>
    
    | 23 |
    
      <a href="../../../blog/page/24/">Next &raquo;</a>
    
  </div>


    </div>

    <nav class="navbar navbar-expand-sm navbar-light fixed-bottom bg-light">
        <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#external-links" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
            <span class="navbar-toggler-icon"></span>
        </button>
        <div class="collapse navbar-collapse" id="external-links">
            <ul class="navbar-nav mr-auto">
                
                <li class="nav-item">
                    <a class="nav-link" href="https://www.linkedin.com/in/ericmjl"><i class="fa fa-linkedin" aria-hidden="true"></i> LinkedIn</a>
                </li>
                
                <li class="nav-item">
                    <a class="nav-link" href="http://twitter.com/ericmjl"><i class="fa fa-twitter" aria-hidden="true"></i> Twitter</a>
                </li>
                
                <li class="nav-item">
                    <a class="nav-link" href="http://github.com/ericmjl"><i class="fa fa-github" aria-hidden="true"></i> GitHub</a>
                </li>
                
                <li class="nav-item">
                    <a class="nav-link" href="http://stackoverflow.com/users/1274908/ericmjl"><i class="fa fa-stack-overflow" aria-hidden="true"></i> Stack Overflow</a>
                </li>
                
                <li class="nav-item">
                    <a class="nav-link" href="http://shortwhale.com/ericmjl"><i class="fa fa-envelope-o" aria-hidden="true"></i> Contact Me</a>
                </li>
                
            </ul>
        </div>
    </nav>
    <!-- Boostrap JS imports -->
    <script src="https://code.jquery.com/jquery-3.2.1.slim.min.js" integrity="sha384-KJ3o2DKtIkvYIK3UENzmM7KCkRr/rE9/Qpg6aAZGJwFDMVNA/GpGFF93hXpG5KkN" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.11.0/umd/popper.min.js" integrity="sha384-b/U6ypiBEHpOf/4+1nzFpr53nxSS+GLCkfwBdFNTxtclqqenISfwAzpKaMNFNmj4" crossorigin="anonymous"></script>
    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0-beta/js/bootstrap.min.js" integrity="sha384-h0AbiXch4ZDo7tp9hKZ4TsHbi047NrKGLO3SEJAg45jXxnGIfYzk4Si90RDIqNm1" crossorigin="anonymous"></script>

</body>
