<!doctype html>

<head>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta charset="utf-8">
    <!-- Bootstrap v4beta Imports -->
    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0-beta/css/bootstrap.min.css" integrity="sha384-/Y6pD6FV/Vv2HJnA6t+vslU6fwYXjCFtcEpHbNJ0lyAFsXTsjBbfaDjzALeQsN6M" crossorigin="anonymous">

    <style media="screen">
        body {
            padding-top: 70px;
            padding-bottom: 70px;
        }
    </style>

    <!-- Syntax Highlighter. Use both pygments and hl.js. -->
    <link rel="stylesheet" href="../../../static/pygments.css">
    <link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/default.min.css">
    <script src="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/highlight.min.js"></script>
    <script>hljs.initHighlightingOnLoad();</script>

    <!-- Google Analytics -->
    <script>
        (function(i, s, o, g, r, a, m) {
            i['GoogleAnalyticsObject'] = r;
            i[r] = i[r] || function() {
                (i[r].q = i[r].q || []).push(arguments)
            }, i[r].l = 1 * new Date();
            a = s.createElement(o),
                m = s.getElementsByTagName(o)[0];
            a.async = 1;
            a.src = g;
            m.parentNode.insertBefore(a, m)
        })(window, document, 'script', 'https://www.google-analytics.com/analytics.js', 'ga');

        ga('create', 'UA-12498603-2', 'auto');
        ga('send', 'pageview');
    </script>

    <!-- ClustrMaps Tracking -->
    <!-- <script type="text/javascript" id="clstr_globe" src="//cdn.clustrmaps.com/globe.js?d=nhKoDpoTjWz4pC6CwI-fSy4hPoJ1uXwTLCfMCT3OK_8"></script> -->

    <!-- FontAwesome embed -->
    <script src="https://use.fontawesome.com/cb9dbe8e41.js"></script>

    <!-- Custom CSS -->
    <link rel="stylesheet" href="/static/css/custom.css">
</head>

<title>Blog - Eric J. Ma's Personal Site</title>
<body class="body">
    <nav class="navbar navbar-expand-sm navbar-light fixed-top bg-light">
        <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#local-links" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
            <span class="navbar-toggler-icon"></span>
        </button>
        <div class="collapse navbar-collapse" id="local-links">
            <ul class="navbar-nav mr-auto">
                
                <li class="nav-item">
                    <a class="nav-link" href="/"><i class="fa fa-home" aria-hidden="true"></i> Home</a>
                </li>
                
                <li class="nav-item">
                    <a class="nav-link" href="/resume"><i class="fa fa-file-text" aria-hidden="true"></i> Resume</a>
                </li>
                
                <li class="nav-item">
                    <a class="nav-link" href="/blog"><i class="fa fa-rss" aria-hidden="true"></i> Blog</a>
                </li>
                
                <li class="nav-item">
                    <a class="nav-link" href="/open-source"><i class="fa fa-code" aria-hidden="true"></i> Open Source</a>
                </li>
                
                <li class="nav-item">
                    <a class="nav-link" href="/projects"><i class="fa fa-briefcase" aria-hidden="true"></i> Projects</a>
                </li>
                
                <li class="nav-item">
                    <a class="nav-link" href="/talks"><i class="fa fa-microphone" aria-hidden="true"></i> Talks</a>
                </li>
                
                <li class="nav-item">
                    <a class="nav-link" href="/teaching"><i class="fa fa-university" aria-hidden="true"></i> Teaching</a>
                </li>
                
            </ul>
        </div>
    </nav>
    <div class="container">
        
  
    
  <!-- <div class="blog-post"> -->
  <link rel="stylesheet" href="../../../static/pygments.css">
  <div>
  
    <h1><a href="../../../blog/2016/12/1/the-renaissance-researcher/">The Renaissance Researcher</a></h1>
  
  <p class="meta">
    written by
    
      <a href="https://twitter.com/ericmjl">Eric J. Ma</a>
    
    on 2016-12-01
  </p>
  <p>This is the computational biologist who knows how to design, execute, and analyze wet lab experiments and the generated data. This is the wet lab scientist who knows how to develop software that crunches the massive amounts of data that her experiment generates. Does this person exist?</p>
<p>I'm going to make a bold statement, and then a bold prediction.</p>
<p>This person does not exist. And this person will become huge in demand in the biosciences in the future. And yet this person will turn out to be a team.</p>
<p>I think I'm qualified to make this statement. I've been on both sides of this dichotomy. I was a bench synthetic biologist, failed badly and went into computational evolutionary ecology, and then came back to the bench trying to marry together high throughout experimentation and deep learning. It's really, really tough to handle both at the same time.</p>
<p>Part of it is a matter of context switching. Computation really requires continuous stretches of iterating between coding and thinking. Experiments, by contrast, represent a continuous disruption to focus time. It’s impossible to juggle both in parallel. Moreover, both require month-long stretches in order to have momentum, during which one or the other gets lost. If I code, I lose momentum with my experiments. If I run experiments, I lose momentum with my code. There’s either a cognitive or energy cost to juggling both at a high performance level.</p>
<p>For me, it's a temperament issue. I learned during my three computation years developing algorithms that the quick turnover time of computation, to go from hypothesis to falsifying results, was something that I absolutely suited my temperament. I absolutely dreaded, and had no patience for the prospect of a failed hypothesis needing weeks to verify.</p>
<p>I also work best in "staggered sequential mode", not "parallelized mode". Staggered sequential means as the results of an experiment become easier to forecast, I start something new. Parallelized mode means juggling things together even when the results aren't easily forecastable. There's a subtle distinction there. Parallelized mode doesn't work for me, because I get very easily confused between project details.</p>
<p>With computation, that turnover time is often on the order of dozens of minutes max. With experiments, its weeks. With my temperament, I would much prefer the quick-turnover computational hypothesis testing rather than painstaking and slow rounds of hypothesis testing.</p>
<p>Yet, as a computation type, I face the dilemma all the time: I have a hypothesis for which I have no data to test it with. This means going back into the lab, which means confronting the very pace that my temperament would prefer not to face. Alternatively, I can find people with whom I can collaborate.</p>
<p>And that brings me to my point. I don't think there's such thing as the renaissance researcher. Rather, I think there is such a thing as a renaissance research team. Here, hypothesis generation and testing are a shared activity, and and the data generation and analysis are carried out by respective specialists. It's iterative, and it's undoubtedly going to be slower (at least initially) compared to going solo. But the rewards, I think, are going to be much greater once the team is in the state of ‘flow’.</p>
<p>In some ways, it reminds me of the problem with finding the ideal “data scientist” - one who could wrangle big amounts of data, make plots, infer conclusions, create interactive dashboards, and communicate the results to business executives. Nonsense, this unicorn doesn’t exist, and the industry quickly figured that out, thankfully, employing data science teams comprised of people with complementary skills.</p>

  </div>

    <p><i>Did you enjoy this blog post? <a href="../../../blog/2016/12/1/the-renaissance-researcher/#disqus">Let's discuss more</a>!</i></p>
    <hr class="fancy">
  
    
  <!-- <div class="blog-post"> -->
  <link rel="stylesheet" href="../../../static/pygments.css">
  <div>
  
    <h1><a href="../../../blog/2016/11/10/book-thoughts-the-theory-that-would-not-die/">Book Thoughts: The Theory That Would Not Die</a></h1>
  
  <p class="meta">
    written by
    
      <a href="https://twitter.com/ericmjl">Eric J. Ma</a>
    
    on 2016-11-10
  </p>
  <p>One of two eBooks I bought to read on this trip to Vienna was "The Theory That Would Not Die". It was a good read, and I definitely have come to appreciate more the history behind Bayes' rule and its multiple deaths and resurrections.</p>
<p>Three things stuck out for me.</p>
<p>Firstly, I didn't realize how big of a genius Laplace was, and how prior to the two great wars, the epicentre of scientific inquiry was in France. Laplace basically touched a wide variety of foundational topics that later generations have built on, and his mastery of concepts as diverse as probability, calculus and the likes is amazing. I definitely can appreciate how he was able to connect many disciplines together.</p>
<p>Secondly, many of the great breakthroughs of the past came from people dabbling in a problem for over a decade. Compare that to the expected constant output by professional scientists now. It seems to me that it is a truism that scientific inquiry can only really be conducted under certain circumstances, and funding crunches, KPIs and publication metrics, and distracted service to an academy aren't the ways to get it done.</p>
<p>Finally, the role of the military was such a catalyst for new advances! Yet the military's role in research results in a paradox. We all want new breakthroughs to be open, freely accessible, and used for peaceful, civilian purposes. Yet the military provides the catalytic impetus for accelerating new technologies and ideas for practical use; not even capitalism has that force. The book highlighted how Bayes remained highly theoretical and impractical in the academic realm for decades after the military already started pushing through practical applications of it.</p>
<p>All in all, good book, highly recommended to read!</p>

  </div>

    <p><i>Did you enjoy this blog post? <a href="../../../blog/2016/11/10/book-thoughts-the-theory-that-would-not-die/#disqus">Let's discuss more</a>!</i></p>
    <hr class="fancy">
  
    
  <!-- <div class="blog-post"> -->
  <link rel="stylesheet" href="../../../static/pygments.css">
  <div>
  
    <h1><a href="../../../blog/2016/11/10/vienna-and-imed-2016-reflections/">Vienna &amp; IMED 2016 Reflections</a></h1>
  
  <p class="meta">
    written by
    
      <a href="https://twitter.com/ericmjl">Eric J. Ma</a>
    
    on 2016-11-10
  </p>
  <p>This trip &amp; conference was a productive one! I'm glad I went for it and didn't have to miss out on much (even though WebJet/Turkish Airlines (still can't sort out which) messed up my initial flight booking big-time).</p>
<p>My biggest takeaways from IMED 2016 are as follows:</p>
<ol>
<li>One Health is a big concept for disease surveillance, and I'm convinced it's the right way to do it. Sample not just humans reactively, but also animals proactively.</li>
<li>There's a project to "whole genome sequence" viruses from a whole variety of animals, called the Global Virome Project. It'll likely have a longer-term impact on disease forecasting than virus-specific surveillance networks.</li>
<li><a href="http://www.ecohealthalliance.org/">EcoHealth Alliance</a> (NYC) and <a href="http://metabiota.com/">Metabiota</a> (SF) are doing cool stuff for disease surveillance. </li>
<li>Made some new friends at the conference. Global conferences are always fun.</li>
<li>Learned from Drew (of <a href="http://isid.org/">ISID</a>) that conference locations should also coincide with off-peak tourist timing and with commitments over multiple years; it helps build the case for city sponsorship as well. </li>
<li>Hackathons are fun, though I remain convinced that their main utility is as a catalyst for launching new long-term collaboration efforts, not as a way to advance new ideas.</li>
<li>Receptions at fancy places (we did the Mayor's hall) are a great way to meet new people. Note to self, if I ever organize a conference myself.</li>
</ol>
<p>And then there's Vienna. Oh Vienna, what a beautiful old city you are. (Your donairs are awesome, btw.) I took 2 "tourist days" to walk around the city, and get a feel for it.</p>
<p>I got to do some more iPhone photography, which was a nice change of pace. After finding my way to the Danube, I walked back and took the train to the Vienna city centre. The photos I took were mostly of the architecture, the parks, and activity there. Naturally, I couldn't resist touching up the photos on my computer. Here's <a href="https://goo.gl/photos/Fn1W1zQ4CbWJM8c79">an album</a> of my favourites.</p>
<p>I noticed that the public transit is amazingly advanced. Amazing, all round.</p>
<p>That evening, though, was when I felt the stomach bug rumbling, and I realized I had food poisoning. I'm not sure where I got it, though, as it seemed too soon after trying out Schnitzel for the first time; I thought it might have been my body reacting to an overly-oily dinner.</p>
<p>As such, I spent the second day stroking the cat and catching up on YouTube videos and eBook reading and sleeping in bed. The donair from a store nearby made for a good breakfast though!</p>
<p>That marks the end of a really fun trip to Vienna. Hope to visit again! And I hope to see the others I met at IMED again!</p>

  </div>

    <p><i>Did you enjoy this blog post? <a href="../../../blog/2016/11/10/vienna-and-imed-2016-reflections/#disqus">Let's discuss more</a>!</i></p>
    <hr class="fancy">
  
    
  <!-- <div class="blog-post"> -->
  <link rel="stylesheet" href="../../../static/pygments.css">
  <div>
  
    <h1><a href="../../../blog/2016/11/6/why-phenotypic-interpretation-matters/">Why Phenotypic Interpretation Matters</a></h1>
  
  <p class="meta">
    written by
    
      <a href="https://twitter.com/ericmjl">Eric J. Ma</a>
    
    on 2016-11-06
  </p>
  <p>At IMED 2016, I heard a lot of talks surrounding surveillance efforts. Most of it has been syndromic surveillance, because that's what's currently collectable. For digital disease surveillance, search queries (a marker of "interest" in a disease) are used as a proxy for probable presence of a disease.</p>
<p>These are good for real-time surveillance, but I think it's not good enough from the standpoint of accuracy and understanding mechanism.</p>
<p>For accuracy, syndromic surveillance and search queries are n<sup>th</sup>-degree manifestations of underlying causes. Multiple different pathogens can present the same symptoms. Upticks in searches for a given symptom or pathogen may have multiple underlying causes, though "hearing of the disease" and "being concerned about it" may be related to certain diagnoses and local spread of information.</p>
<p>Found data can help act as a very early sentinel to a problem, and can guide us by providing leads to follow-up, but for understanding mechanism, neither tell us how a new virus is going to be dangerous.</p>
<p>In other words, while fast and rapid and local, syndromic surveillance and digital disease surveillance remains a reactive endeavour, because we can't mechanistically forecast risk from the data.</p>
<p>On the other hand, genomic data is often collected with the intention of understanding pathogen dynamics, as we have well-developed phylogenetic tools for this. However, we still rely on single point mutations, identified in small-scale, one-off, ad-hoc, [insert-some-dashed-phrase-that-expresses-frustration] studies, and basically treat the limited amount of evidence as gospel. I am quite sure that is not the right way to do phenotypic interpretation of genomic data.</p>
<p>If we're going to make disease surveillance a <strong>proactive</strong> endeavour, I am convinced that the disease surveillance world will need a paradigm shift, and investment in infrastructure. By predictive, this is what I mean:</p>
<ol>
<li>Calculating what has a high probability of jumping into humans.</li>
<li>Phenotypically testing those viruses without using live viruses (concept of pseudotyping).</li>
<li>Using genomic information to predict phenotype.</li>
</ol>
<p>Here's how I see it happening, given everything I've learned at IMED 2016:</p>
<p><strong>(1) Sequence everything, in animals that interface with humans</strong></p>
<p>60-70% of new infections are going to be zoonotic in nature. To know what might be coming up, use metagenomics to sequence animal species in geographic areas that humans, <em>prior</em> to the host switch event. To me, this is a no-brainer. Long-term serial sampling is going to be key.</p>
<p><strong>(2) Leverage pseudotyping technologies and genetic systems, and automation, to create systematic phenotyping platforms</strong></p>
<p>Doing so will allow us to phenotypically test components of a virus for a particular step in its life cycle, such as speed of entry into a cell, or resistance to a drug. With rapid DNA synthesis &amp; assembly technologies, we should be able to experimentally measure a virus' epidemiologically-relevant phenotype within 24 hours of an outbreak. We should also be able to rapidly generate large-scale genotype-phenotype data for machine learning.</p>
<p>What would be relevant phenotypes? I can think of a few, broken down basically to "druggability" and "life cycle steps":</p>
<ol>
<li>For druggable proteins, measure drug resistance. It has high epidemiological relevance, and hence, value.</li>
<li>For viruses targetable by drugs and not alike, measure cellular entry capability.</li>
<li>For identified polymerase complexes, measure replication capacity.</li>
<li>For cellular release machinery, if identified, measure release.</li>
<li>If there's others already existing for which the case can be made for systematic phenotyping, scale the assay up.</li>
</ol>
<p><strong>(3) Develop machine learning tools that allow for learning of structural properties that determine epi-relevant protein phenotype</strong></p>
<p>By doing so, we can explicitly model epistasis in a protein, and therefore account for epistatic interactions explicitly when developing learning algorithms that map genotype to phenotype. With this kind of technology coupled to the ability to sequence a virus within hours of sample collection, we'll be able to cut down the time for phenotypic interpretation from the order of a day (because of experimentation required) to seconds (because it's a computational prediction). By going Bayesian, we can even model uncertainty in final predictions, thereby aiding decision-making under uncertainty.</p>
<hr>
<p>In conjunction with the "sequence everything" approach, systematic phenotyping and deep learning will allow us to move to a "predictive" paradigm because it'll give us the ability to rationally predict the phenotypic risk of a virus before it even jumps into humans. I see this as an infrastructural project: requiring up-front commitment of time and money. Yet, we know that investments in infrastructure always pay off dividends in a myriad of ways, from time saved to lives saved.</p>
<p>Something I learned from IMED 2016, however, is that epidemiology is an old discipline, and with old disciplines comes a certain tunnel vision on how things should be done. It'll be up to a new and creative generation (i.e. us) to convince the funders and old guard that our new way will add value to the discipline.</p>

  </div>

    <p><i>Did you enjoy this blog post? <a href="../../../blog/2016/11/6/why-phenotypic-interpretation-matters/#disqus">Let's discuss more</a>!</i></p>
    <hr class="fancy">
  
    
  <!-- <div class="blog-post"> -->
  <link rel="stylesheet" href="../../../static/pygments.css">
  <div>
  
    <h1><a href="../../../blog/2016/11/5/imed-2016-talks/">IMED 2016 - Talks</a></h1>
  
  <p class="meta">
    written by
    
      <a href="https://twitter.com/ericmjl">Eric J. Ma</a>
    
    on 2016-11-05
  </p>
  <p>Today (Day 1) was the day that talks and posters really started. Hackathon participants missed the first day, in which there was a plenary session of "high-level speakers". A bit of a pity to miss it, but that's okay - the hackathon was more fun.</p>
<p>After delivering the Animal Village hackathon pitch once more in the morning with my teammate Emily Iacobucci, I hung around to learn more about what people in the Emerging Infectious Diseases &amp; Surveillance community are thinking about. I was most interested in what set of tools were being developed, and what problems they solved. Below are my summarized notes.</p>
<p><strong>Talks</strong></p>
<p>For the talks, I focused mostly on the experimental and data collection tools being developed. Also attended one talk on forecasting new emerging diseases - this is like the holy grail of infectious diseases - can we predict where new outbreaks are going to happen? Also in development are genetic tools to manipulate <em>any</em> virus. Key tools I took note of included:</p>
<ul>
<li>Number of different viral hosts is a great predictor of spillover into humans. | <a href="http://www.nature.com/articles/srep14830">paper</a></li>
<li>Genetic systems for <a href="https://en.wikipedia.org/wiki/Pseudotyping">pseudotyping</a>. | <a href="https://www.ncbi.nlm.nih.gov/pubmed/25936665">paper (paywall)</a></li>
<li>Syndromic and/or search surveillance software and participatory community surveillance by patients/lay people (<a href="http://www.skollglobalthreats.org/tag/podd/">PODD</a>/<a href="https://flunearyou.org/">Flu Near You</a>) and physicians (<a href="http://www.uptodate.com/home">UpToDate</a>)</li>
</ul>
<p><strong>Posters</strong></p>
<p>For the posters, I mainly caught three posters that dealt with data and modelling; most of the other posters were about some specific outbreak in some specific animal or some specific region. These are all good and such; it's just that my nature inclines me towards generalizable topics:</p>
<ul>
<li>Validation of NA-Fluor assay for clinical use: most interested because of genotype-phenotype problem I'm tackling.</li>
<li>Data sharing by the <a href="https://www.iddo.org/">International Diseases Data Observatory</a>: an interesting one where I learned that data governance has to be done right to get people to share data.</li>
<li>Modelling of disease spread using airplane travel data, by <a href="http://www.ecohealthalliance.org/">EcoHealth Alliance</a></li>
</ul>
<hr>
<p><strong>Update from Day 2</strong></p>
<p>The talks that I picked to hear today were centred mostly on data and antimicrobial resistance (AMR). Highlights include the following:</p>
<ol>
<li>European CDC has a network of AMR surveillance efforts that are going on. I think a lot of it has to be organically grown in individual nations, and can't be forced upon.</li>
<li>Data sharing ethics: Representatives from the Wellcome Trust, WHO, and European CDC gave an overview of the principles behind best open data sharing practices, issues and challenges currently faced, and ethics.</li>
<li>Principles for data sharing:<ol>
<li>Using data to help communities be prepared for an outbreak.</li>
<li>Speed but not at the cost of issues of "justice" (e.g. not inadvertently marginalizing groups in interest of speed)</li>
<li>Identifying appropriate level of risk that society can bear.</li>
<li>Building relationships and confidence.</li>
<li>Moving focus away from ownership of data to providing a fair and just system for data recognition.</li>
</ol>
</li>
</ol>
<p>Apart from this, I sat in on Eddy Rubin's talk on the global virome (more like "stood in", it was standing room only). I am convinced that taking a "sequencing everything" approach is likely going to be the key for predictive surveillance. The next step is whether we'll be able to interpret risk from the genome or not, but that'll be the focus of my next post.</p>

  </div>

    <p><i>Did you enjoy this blog post? <a href="../../../blog/2016/11/5/imed-2016-talks/#disqus">Let's discuss more</a>!</i></p>
    <hr class="fancy">
  

  
  <div class="pagination">
    
      <a href="../../../blog/page/13/">&laquo; Previous</a>
    
    | 14 |
    
      <a href="../../../blog/page/15/">Next &raquo;</a>
    
  </div>


    </div>

    <nav class="navbar navbar-expand-sm navbar-light fixed-bottom bg-light">
        <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#external-links" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
            <span class="navbar-toggler-icon"></span>
        </button>
        <div class="collapse navbar-collapse" id="external-links">
            <ul class="navbar-nav mr-auto">
                
                <li class="nav-item">
                    <a class="nav-link" href="https://www.linkedin.com/in/ericmjl"><i class="fa fa-linkedin" aria-hidden="true"></i> LinkedIn</a>
                </li>
                
                <li class="nav-item">
                    <a class="nav-link" href="http://twitter.com/ericmjl"><i class="fa fa-twitter" aria-hidden="true"></i> Twitter</a>
                </li>
                
                <li class="nav-item">
                    <a class="nav-link" href="http://github.com/ericmjl"><i class="fa fa-github" aria-hidden="true"></i> GitHub</a>
                </li>
                
                <li class="nav-item">
                    <a class="nav-link" href="http://stackoverflow.com/users/1274908/ericmjl"><i class="fa fa-stack-overflow" aria-hidden="true"></i> Stack Overflow</a>
                </li>
                
                <li class="nav-item">
                    <a class="nav-link" href="http://shortwhale.com/ericmjl"><i class="fa fa-envelope-o" aria-hidden="true"></i> Contact Me</a>
                </li>
                
            </ul>
        </div>
    </nav>
    <!-- Boostrap JS imports -->
    <script src="https://code.jquery.com/jquery-3.2.1.slim.min.js" integrity="sha384-KJ3o2DKtIkvYIK3UENzmM7KCkRr/rE9/Qpg6aAZGJwFDMVNA/GpGFF93hXpG5KkN" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.11.0/umd/popper.min.js" integrity="sha384-b/U6ypiBEHpOf/4+1nzFpr53nxSS+GLCkfwBdFNTxtclqqenISfwAzpKaMNFNmj4" crossorigin="anonymous"></script>
    <script src="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0-beta/js/bootstrap.min.js" integrity="sha384-h0AbiXch4ZDo7tp9hKZ4TsHbi047NrKGLO3SEJAg45jXxnGIfYzk4Si90RDIqNm1" crossorigin="anonymous"></script>

</body>
