<!doctype html>

<head>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta charset="utf-8">
    <!-- Bootstrap v4beta Imports -->
    <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.1.1/css/bootstrap.min.css" integrity="sha384-WskhaSGFgHYWDcbwN70/dfYBj47jz9qbsMId/iRN3ewGhXQFZCSftd1LZCfmhktB" crossorigin="anonymous">

    <style media="screen">
        body {
            padding-top: 70px;
            padding-bottom: 70px;
        }
    </style>

    
    

    <!-- Google Analytics -->
    <script>
        (function(i, s, o, g, r, a, m) {
            i['GoogleAnalyticsObject'] = r;
            i[r] = i[r] || function() {
                (i[r].q = i[r].q || []).push(arguments)
            }, i[r].l = 1 * new Date();
            a = s.createElement(o),
                m = s.getElementsByTagName(o)[0];
            a.async = 1;
            a.src = g;
            m.parentNode.insertBefore(a, m)
        })(window, document, 'script', 'https://www.google-analytics.com/analytics.js', 'ga');

        ga('create', 'UA-12498603-2', 'auto');
        ga('send', 'pageview');
    </script>

    <!-- ClustrMaps Tracking -->
    <!-- <script type="text/javascript" id="clstr_globe" src="//cdn.clustrmaps.com/globe.js?d=nhKoDpoTjWz4pC6CwI-fSy4hPoJ1uXwTLCfMCT3OK_8"></script> -->

    <!-- FontAwesome embed -->
    <!-- <script src="https://use.fontawesome.com/cb9dbe8e41.js"></script> -->
    <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.1.1/css/all.css" integrity="sha384-O8whS3fhG2OnA5Kas0Y9l3cfpmYjapjI0E4theH4iuMD+pLhbf6JI0jIMfYcK3yZ" crossorigin="anonymous">

    <!-- Custom CSS -->
    <link rel="stylesheet" href="/static/css/custom.css">

    <!-- Mathjax -->
    <script type="text/javascript" async
        src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/latest.js?config=TeX-MML-AM_CHTML">
    </script>

    <script type="text/x-mathjax-config">
        MathJax.Hub.Config({tex2jax: {inlineMath: [['$','$'], ['\\(','\\)']]}});
    </script>

</head>

<title>Blog - Eric J. Ma's Personal Site</title>
<body class="body">
    <nav class="navbar navbar-expand-sm navbar-light fixed-top bg-light">
        <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#local-links" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
            <span class="navbar-toggler-icon"></span>
        </button>
        <div class="collapse navbar-collapse" id="local-links">
            <ul class="navbar-nav mr-auto">
                
                <li class="nav-item">
                    <a class="nav-link" href="/"><i class="fa fa-home" aria-hidden="true"></i> Home</a>
                </li>
                
                <li class="nav-item">
                    <a class="nav-link" href="https://ericmjl.github.io/resume"><i class="fa far fa-file-alt" aria-hidden="true"></i> Resume</a>
                </li>
                
                <li class="nav-item">
                    <a class="nav-link" href="/blog"><i class="fa fa-rss" aria-hidden="true"></i> Blog</a>
                </li>
                
                <li class="nav-item">
                    <a class="nav-link" href="/open-source"><i class="fa fa-code" aria-hidden="true"></i> Open Source</a>
                </li>
                
                <li class="nav-item">
                    <a class="nav-link" href="/projects"><i class="fa fa-briefcase" aria-hidden="true"></i> Projects</a>
                </li>
                
                <li class="nav-item">
                    <a class="nav-link" href="/talks"><i class="fa fa-microphone" aria-hidden="true"></i> Talks</a>
                </li>
                
                <li class="nav-item">
                    <a class="nav-link" href="/teaching"><i class="fa fa-university" aria-hidden="true"></i> Teaching</a>
                </li>
                
                <li class="nav-item">
                    <a class="nav-link" href="/bio"><i class="fa fa-user" aria-hidden="true"></i> Bio</a>
                </li>
                
            </ul>
        </div>
    </nav>
    <div class="container">
        
  
    
  <!-- Import stylesheet -->
  <link rel="stylesheet" href="../../../static/pygments.css">
  <div>

  <!-- Set title style -->
  
    <h1><a href="../../../blog/2016/4/7/the-back-story-to-our-publication-on-influenza-reassortment/">The back-story to our publication on influenza reassortment.</a></h1>
  

  <!-- Append author -->
  <p class="meta">
    written by
    
      <a href="https://twitter.com/ericmjl">Eric J. Ma</a>
    
    on 2016-04-07
  
  </p>
  <!-- Append tags after author -->
  <p>
      <i class="fas fa-tag"></i>
      <a href="../../../blog/tag/science/">
          science
      </a>
      <i class="fas fa-tag"></i>
      <a href="../../../blog/tag/graduate school/">
          graduate school
      </a>
      <i class="fas fa-tag"></i>
      <a href="../../../blog/tag/influenza/">
          influenza
      </a>
      <i class="fas fa-tag"></i>
      <a href="../../../blog/tag/peer review/">
          peer review
      </a>
      
  </p>
  

  <!-- Put post body -->
  <p>For the reader of a newly-published article, all that we see is precisely that - the article itself. We rarely get to hear about the back-story of that paper, or the choices that were made, the struggles involved, and the emotional ride taken. I thought I'd take the time to document what the back-story to our manuscript on <a href="http://biorxiv.org/content/early/2016/03/22/033514">influenza reassortment</a> was like. Hopefully, it'll let other junior trainees know that nobody's alone in the struggle.</p>
<p>Now, where shall I start...?</p>
<p>I think the best part to start is when we first conceived the idea. My advisor, Jon Runstadler, had just taken me on. Because I joined only at the end of my 2nd year, I had some catching up to do.</p>
<p>The problem of influenza reassortment had started to catch my eye. From my reading of the literature, reassortment was implicated in all known pandemics. The problem of predicting reassortment sounded like something impactful (in the sense of being practical, not prestigious) that I could dedicate my time to. I also happened to learn about the Influenza Research Database, which housed a large dataset of sequenced influenza genomes. I thought, "How great might it be if we could identify every reassortant virus in the database? We could possibly predict the next pandemic!" In retrospect, it's a naive idea to think that we can identify every single one of them accurately and use that data alone to predict pandemics. But my lack of knowledge on phylogenetics, reassortment, and influenza biology gave me enough ignorance to knuckle on. (I happen to believe that sufficient ignorance is necessary for creative breakthroughs.)</p>
<p>Fast-forward two years later, the following things happen:</p>
<ol>
    <li><strong>Committee Meeting 1:</strong> I get a lecture from my committee about the importance of simulated data for a computational study. I was a complete newcomer to computational research at that point, so it was a sorely needed reminder.</li>
    <li><strong>October 2013:</strong> <a href="https://sph.uth.edu/justinbahl/">Dr. Justin Bahl</a>, an expert in the use of Bayesian phylogenetic methods, gives us a masterclass on how to use BEAST, and I bounce some of my ideas off him. Huge learning opportunity!</li>
    <li><strong>December 2013:</strong> After hundreds of hours of debugging, I hack some <a href="https://d3js.org/">d3.js</a> code to visualize reassortment! We get some nice bouncy reassortment + clonal transmission trees. And then I vow to avoid Javascript as much as possible...</li>
    <li><strong>June 2014:</strong> After hundreds of hours of debugging, I manage to reproduce, using our reassortment detection method, known reassortant viruses. Hooray!</li>
    <li><strong>September 2014:</strong> After hundreds of hours of debugging, I finally manage to get my code running on a download of the IRD. And then I figure out that I need better software engineering skills...</li>
    <li><strong>December 2014:</strong> With little debugging needed, we manage to show that the method we're using essentially is a very, very good approximation to a phylogenetic reconstruction. Ooh yeah!</li>
    <li><strong>February 2015:</strong> Jon suggests that I start writing up a manuscript. I re-run the code on a fresh download of the database, with some refactoring done, and it still works! At this point, I still don't know what the story is...</li>
    <li><strong>June 2015:</strong> After some exploratory analysis, I realize that the dataset on hand and the identification of reassortant viruses is an awesome grounding for an ecological study of influenza evolution. That becomes the story that we write up.</li>
</ol><hr />

<p><h3>Let the Submissions Begin</h3>
As I was writing, I had in mind a number of target journals for the paper. Because of the nature of the story, and balancing the desire to go open access, I had a vague shortlist that included a group of microbiology, ecology, and genetics journals. But when I discussed it with Jon, the first place we were going to submit the article to was revealed to me.</p>
<p><em><a href="http://www.nature.com/index.html">Nature</a>.</em></p>
<p>Alrighty... The stakes are raised!</p>
<p>I go about formatting the paper for Nature. It takes about 1 month to check that everything is in order, including the cover letter, the figures, the Jupyter notebooks. One afternoon, in Jon's office, we submit the paper together.</p>
<p>11 days later, Nature editorially declines the manuscript.</p>
<p>Looking back on Jon's forwarded email to me, it's the best words I could have heard given that this was the first rejection manuscript rejection letter I had ever received. <em>"Well, it's their loss."</em></p>
<p>In the declination letter, the Nature editors inform us that a new journal, <a href="http://www.nature.com/nmicrobiol/">Nature Microbiology</a>, is in the process of being setup. Since it's a convenient option, and the editors suggested it, we go ahead with the option.</p>
<p>We have a quick discussion on a backup journal. I suggest <a href="https://elifesciences.org/">eLife</a>, because they're a new journal, is open-access, has broad readership, supports new researchers (e.g. with reference letters), doesn't have nit-picky formatting requirements, and all-in-all had a great up-and-coming reputation. We agree to go with that if Nature Microbiology rejects.</p>
<p>8 days later, Nature Microbiology editorially declines the manuscript.</p>
<p>I spend another week or two reformatting for eLife, and getting everything in order.</p>
<p>4 days later, eLife contacts us!</p>
<p>Their editors <em>actually read through the whole paper</em>, and had concerns about sampling biases, and asked for specific analyses as a check. (This incidentally re-kickstarted a long discussion in our lab about how to deal with sampling biases, which as far as I can see, has been largely ignored by the influenza phylogeneticists.) As I'm flying around on conferences, I end up needing about a month to get everything in order, and confer with Jon long-distance. There's a lot of thinking, debating, arguing that goes on in the wee hours of the night after conference stuff. (I think I was at SciPy 2015 at this point.) After re-submitting the manuscript to them, we wait.</p>
<p>1 day later, eLife's editors respond back.</p>
<p><em>"Man, this stings..."</em> I remember thinking that when I saw the decision letter. In the heat of the moment, I remember feeling like I was punched in the stomach. Was all that effort for naught? What did we do wrong this time round?</p>
<p>In retrospect, and in fairness to their efforts, eLife's editors were the only ones I've seen that, despite rejecting the manuscript, actually gave our's a fair reading. They even went into my Github repository and crunched numbers. For me, that's earned my respect, and I still would highly recommend them as a top destination journal. Jon also mentioned that it was a good sign that the editors took the manuscript seriously enough to consider it and read it in-depth. We interpreted this as a sign of general interest. (If the eLife editors read this, I am happy to be corrected on this thought.) <em>Forge on.</em></p>
<p>After this rejection, I conferred with Jon a bit, and he's still convinced that it should go to a broad readership journal. I take a day off to get a breather, to reflect on the whole thing so far. 3 editorial rejections, it's not a nice feeling. I realize that the eLife editors were really hung up on one of the figures, which really wasn't the main point of the paper. I decide that the paper needs a rewrite, with additional analysis to address the eLife editors' concerns.</p>
<p>I ask Jon for a month to rewrite the manuscript. It's about August/September, and the new academic year is starting. Not that it matters for me at this point; the calendar year is starting to look the same regardless.</p>
<p>In doing the rewrite, I took a look at the four figures in the paper. There was 1a-d, 2a-d, 3a-g, 4a-b. Of all of those, only 3c and 4a were necessary. Everything else was <a href="http://fluffyguy.com/">fluffy-ness</a>. I was on my first phase of weight loss back then, so I think that this "cut the fluff" way of thinking was pre-eminent in my mind. I rewrite the paper, and tried to make it more focused and streamlined, perhaps... <em>essential</em>. I send it to Jon, and after editing, he informs me of his next target destination.</p>
<p><em><a href="http://www.sciencemag.org/">Science</a></em></p>
<p>I make further reworks, condensing it to below 2,500 words, and we upload the manuscript. It's now about the end of November.</p>
<p>A few days later, Science declines the manuscript.</p>
<p>By this point, having gone through the low point of eLife's rejection, and having the badge of honor of being rejected from Nature, I'm starting to get impervious to journal rejections. This one didn't affect my emotions much. I confer again with Jon, and propose a list of backup journals. The only broad readership journal I had on my original list was <a href="http://www.pnas.org/">PNAS</a>, so I proposed that, followed by something along the likes of <a href="http://journals.plos.org/plospathogens/">PLOS Pathogens</a>, <a href="http://www.nature.com/ismej/index.html">ISME Journal</a>, <a href="http://www.cell.com/current-biology/home">Current Biology</a>, <a href="http://www.genetics.org/">Genetics</a>, <a href="http://www.nature.com/hdy/index.html">Heredity</a>, <a href="http://jid.oxfordjournals.org/">Journal of Infectious Disease</a> (I'm going by memory now, and not double-checking any email trace as I did for the above narrative.)</p>
<p>Jon thinks that <a href="http://www.cell.com/cell-host-microbe/home">Cell Host Microbe</a> might be worth a shot. I did some homework on it, and thought it might be good too, but I was reluctant to waste time on reformatting the paper only to be editorially rejected again. Jon then suggests doing a pre-submission inquiry using the abstract.</p>
<p>2 days later, Cell Host Microbe declines.</p>
<p>We go ahead with the plan, and I format everything for PNAS. My hopes aren't high; at this point, I'm just happy to have the paper reviewed somewhere, so that I can get feedback to make it better. That's all I want. Borne out of this frustration and desire, I ask Jon for permission to post the manuscript to BioRxiv, and he agrees. I also run the paper by the BE Communications Lab, to see if it makes sense to a fellow biologist.</p>
<p>Around mid-December, we submit the paper to PNAS.</p>
<p>While waiting, I busy myself with a side project to help with a seal sampling project that's ongoing - hacking together a seal monitoring device using Raspberry Pi computers, Bluetooth LE beacons, and Raspberry Pi camera modules stuck into food containers and mounted on garden poles. For me, it's basically an excuse (ahem, <em>reason</em>) to play with a bunch of Raspberry Pi computers, but doing something useful and contributing to the research group while at it. One day, I'll write a blog post on it.</p>
<p>On 1 Feb 2016, I get an forwarded email from Jon. "This is good. Several things to work on and discuss in the reviewers' comments. Take a look and let's find time to lay out a plan to tackle."</p>
<p>I look through the reviewer's comments. A few points popped out immediately:</p>
<ul>
<li>There's no 3rd reviewer. <em>There's no fabled 3rd reviewer. There's no 3rd reviewer!</em></li>
<li>Reviewer 1 was very detailed, gave very constructive comments on how to improve the clarity of the manuscript, and was instrumental in shaping the extent to which we could claim the conclusions in the paper. Jon's own words: "My faith in the peer review process has been restored."</li>
<li>Reviewer 1 also helped us discover where I left a chunk of the methods out. Very crucial! Because I was getting too familiar with the manuscript, I was skimming over things; my co-authors were also too familiar with the work as well to have picked on it. We had spent tons of time going over and over the paper.</li>
<li>Reviewer 2 was very encouraging, and basically recommended it for publication. Reviewer 2 was brief, but nonetheless gave constructive criticisms on the paper as well.</li>
</ul>
<p>This was so much more than I had expected! I was braced for scathing reviews on sampling biases, comparisons of inadequacy to phylogenetic methods, making over-reaching claims... and had my own dreams (nightmares?) of having to deal with comments on those points yet not being able to find a way to do so. Either way, I was very encouraged by these comments, and promptly bunkered down in the library, Starbucks, and the Broad, writing code and prose to address their review points.</p>
<p>Over the course of the month, I meet with Jon about 2 more times to make sure the responses are on-target. After finishing everything, I also run the response letter past the <a href="https://be.mit.edu/communicationlab">BE Communications Lab</a> again, which helped further in structuring our response to the toughest point of review. (I am happy to report that we have acknowledged their help in the paper.)</p>
<p>On March 9th, we re-submit the paper back to PNAS.</p>
<p>In the intervening time, I start getting nervous again. What if all of this effort was for nothing? What if, even after all the favourable reviews, the editor still says no? What if I bombed the response letter? I try imagining the most probable scenarios, but regardless of the minute probabilities of it happening, none speak emotionally louder than another rejection letter. I try blocking it out. As a distraction, I go up to Harvard and start embarking on applying deep learning to protein graphs from the <a href="http://hips.seas.harvard.edu/">HIPS</a> group instead, learning from a fellow UBC alum &amp; Canadian <a href="https://scholar.google.com/citations?user=ZLpO3XQAAAAJ">David Duvenaud</a>, who is my mentor up there.</p>
<p>On March 30th, the decision comes back. I'm sitting in the middle of a seminar when I have this sudden urge to check my email. There's only one unread message in my Inbox at that time, and it's titled "PNAS MS# 2015-22921R Decision Notification". My heart is thumping, what will it say?</p>
<blockquote><p>March 30, 2016</p>
<p>Title: "Reticulate evolution is favored in influenza niche switching"</p>
<p>Tracking #: 2015-22921R</p>
<p>Authors: Ma et al.</p>
<p>Dear Dr. Ma,</p>
<p>We are pleased to inform you that the PNAS Editorial Board has given final approval of your article for publication...</p>
</blockquote>
<p>Big! Smiles! Everywhere! Jumps! Up! And! Down!</p>
<p>So, I'm not Dr. Ma yet, but hopefully close to there. (If I cross that finish line, I'm opting for an ScD title, just because... I get a yellow hood!) It's a project taken from conception to completion. From start to finish. From scratch to final product. In the short-term, the most important hurdle has been crossed. The feelings that of (firstly) exhilaration, then relief, then pure unbridled joy.</p>
<p>I inform my co-authors by email. Later in the day, I learn that my classmate Daniel Martin-Alarcon also had a publication accepted in PNAS on the exact same day. Double the joy, as we shared the news over lunch.</p>
<hr>

<p>And so that's the back story! All the rejections, the ups and downs, the nervousness, the rejoicing... But to have it end in an acceptance letter is a wonderful result. I am thankful to my advisor <a href="runstadlerlab.mit.edu">Jon</a>, my co-authors <a href="https://www.linkedin.com/in/nichola-j-hill-82917834">Nichola</a>, <a href="https://www.linkedin.com/in/justin-zabilansky-95390b3b">Justin</a> and <a href="https://www.linkedin.com/in/kyle-yuan-98ba6544">Kyle</a>, who helped shape the story, and who contributed code and data. Final thanks to the two reviewers who helped strengthen the logic of the manuscript, who raised thoughtful questions, and pointed out logical and linguistic gaps in the paper.</p>
<p>At this point, it feels as if a large burden has been lifted off my shoulder, and I can take a much needed break and reflect on stuff I've learned. As for that, I will detail that in the next blog post. Meanwhile, I am riding momentum on a newer project that I launched a while back, to predict phenotype from genotype. <strong>Onward!</strong></p>

  </div>

    <p><i>Did you enjoy this blog post? <a href="../../../blog/2016/4/7/the-back-story-to-our-publication-on-influenza-reassortment/#disqus">Let's discuss more</a>!</i></p>
    <hr class="fancy">
  
    
  <!-- Import stylesheet -->
  <link rel="stylesheet" href="../../../static/pygments.css">
  <div>

  <!-- Set title style -->
  
    <h1><a href="../../../blog/2016/3/29/life-after-science-one-a-journey-in-computation-and-creativity/">Life after Science One: A Journey in Computation and Creativity</a></h1>
  

  <!-- Append author -->
  <p class="meta">
    written by
    
      <a href="https://twitter.com/ericmjl">Eric J. Ma</a>
    
    on 2016-03-29
  

  <!-- Put post body -->
  <p><em>Writer's note: This blog post was written for the class of Science One 2015/2016, 9 years on after my own experience in Science One 2006/2007. My classmate, Jacob Bayless, is giving a talk to them titled "Life After Science One", and reached out to me for some perspectives on learning computation. Here's my piece, for him, and for this year's class and beyond.</em></p>
<p>10 years ago, I joined UBC as a student in the Science One program. That year was a fun year, and one of the best educational experiences I've had. During Science One, we learned to think integratively across disciplines. For example, we saw how order of magnitude estimation, a common tool in physics, could be applied to ecology, biochemistry, and thermodynamics. As another example, we learned about the application of ordinary differential equations to ecology and immunology in predator-prey systems. That way of thinking - by bridging disciplines and meshing ideas - is something I've re-discovered, re-encountered, and re-applied over and over in my research career to date. It doesn’t stop, even after Science One.</p>
<p>There was something I wish was emphasized back in my year, which you all now have the privilege of learning: computation and <a href="https://en.wikipedia.org/wiki/Computational_thinking">computational thinking</a>.</p>
<p>While at UBC, I did some quantitative classes, including multivariable calculus, introductory programming (Java was all the rage back then), and statistics, but nothing more than that. By the end of my undergraduate training, I was thoroughly trained in molecular biology, but utterly helpless with programming ecosystems. I only knew how to transcribe and translate DNA sequences in Java.</p>
<p>Later, I transitioned into doing computational work during my PhD. I was ending my 2nd year in the MIT Biological Engineering department, and in search of a good topic to work on for my thesis. I was a 2-month old Pythonista (this is what Python programmers call ourselves, so you all are Pythonistas now!) at that point, and I was teaching myself Python to improve cloning workflows by automating PCR primer design. On the recommendation of a friend, I checked out the Boston Python meetup group. There, I met Giles, a software developer with a gene synthesis company, and a former <a href="http://www.broadinstitute.org">Broadie</a>. I told him this idea I had to classify all of the internal genes of the influenza virus. Looking back, it’s a bad idea to try attempting this for a thesis project, but Giles and I were both naive enough about the problem that we set about talking through it. I drew him a matrix, he came back with an idea, we Googled stuff up, I came up with an idea, and drew another thing.</p>
<p>Rinse, wash, repeat. It was such an energizing and exciting time! Giles looks at our ideas, and says, “I think you need a clustering algorithm. Try… <a href="http://science.sciencemag.org/content/315/5814/972.full">affinity propagation</a>. It’s a relatively new one, but nonetheless has a mature implementation available in Python. Search <a href="http://scikit-learn.org/stable/modules/generated/sklearn.cluster.AffinityPropagation.html"><code>scikit-learn</code></a> for it.” In effect, he was asking a 2-month old Pythonista to do machine learning in Python. Well played, Giles; fast forward a few months, next thing I knew, he had effectively kickstarted the groundwork for my thesis, which would eventually evolve into a <a href="http://biorxiv.org/content/early/2016/03/22/033514">computational study</a> of influenza’s capacity for reticulate evolution (through reassortment) and its importance in switching hosts (or ecological niches). My advisor Jon, though not a computationally trained person himself, trusted me with the freedom to learn, fail, and create under his mentorship, and I am hoping that in due time, we can reap the fruit of this trust.</p>
<p>We ended up never using machine learning in that paper above, but a few things happened that got me deeper into computation. My experience working with the <code>scikit-learn</code> library piqued my interest in machine learning tools as applied to biology. I learned about network analysis through Allen Downey’s book, “Think Complexity”, and incorporated it as my main modelling tool as well. I went to PyCon 2014 and 2015 (in Montreal), giving tutorials on data analysis and network analysis. While on conference, I also learned a ton as well about how to use the <code>scikit-learn</code> libraries, and good practices in the software development world that could be used in the research world. This year, I will be at PyCon 2016 giving a <a href="https://us.pycon.org/2016/schedule/presentation/1578/">tutorial</a> on statistical network analysis as well, while also continuing the learning journey in the Python and data science worlds. The learning journey continues; I soon discovered for myself that scikit-learn alone wasn’t enough, and I’ve started learning the internals of deep learning from a <a href="http://hips.seas.harvard.edu/">research group</a> up at Harvard, with the goal of applying it to developing highly interpretable models of phenotype from genotype. At the same time, the environment here (MIT &amp; the Broad Institute) helps; there's a <a href="http://www.broadinstitute.org/scientific-community/science/mia/models-inference-algorithms">Models, Inferences &amp; Algorithms</a> seminar series, during which we learn both the mathematical underpinnings of computational methods, and their application to biomedical problems.</p>
<p>For the past 9 years, the learning hasn’t stopped. I’ve found that thinking integratively, which is exactly what is taught in Science One, is a crucial element in the creative process. New ideas come from executing on the juxtaposition and composition of old ideas, in ways others have never done before. Computation has also played a huge role in enabling me to tackle problems that would otherwise be beyond my reach; from simple automation to modelling complex problems, it’s a superpower you’ll want to have. Computational thinking has helped me scale the scope of what I could solve. So, to SciOne 15/16, cherish the chance learn what you have learned so far. As soon as you can, find an outlet, a problem on which to apply what you’ve learned; go solve a problem for the world around you, whether big or small. Find people who will teach you, trust you with resources to finish the task, and pick you up when you're down. And while you’re at it, have a ton of fun!</p>

  </div>

    <p><i>Did you enjoy this blog post? <a href="../../../blog/2016/3/29/life-after-science-one-a-journey-in-computation-and-creativity/#disqus">Let's discuss more</a>!</i></p>
    <hr class="fancy">
  
    
  <!-- Import stylesheet -->
  <link rel="stylesheet" href="../../../static/pygments.css">
  <div>

  <!-- Set title style -->
  
    <h1><a href="../../../blog/2016/3/22/webinar/">Webinar</a></h1>
  

  <!-- Append author -->
  <p class="meta">
    written by
    
      <a href="https://twitter.com/ericmjl">Eric J. Ma</a>
    
    on 2016-03-22
  

  <!-- Put post body -->
  <p>Today, I gave a webinar to the IRD and ViPR technical and advisory board meeting. There were a number of challenges to giving a webinar that I want to reflect on here, as a note to my future self and others who may read this entry. In contrast to my previous entry, I'm choosing to write this down fairly soon after doing the presentation, so that I can remember the details clearly.</p>
<p>In the lead-up to the webinar, I felt pretty well-prepared. I had done at least 3-4 rounds of narration on my own, so I was very familiar with the content, transitions, and flow. I also had a set of slides prepared for the Q&amp;A session at the end. What I did not expect, but now have come to learn about, was the challenge of interacting with an audience that I can only hear but not see.</p>
<p>Visual communication is important. During the webinar, because I could not see my audience, I made the mistake of perceiving audience questions as being communicated in an aggressive fashion, rather than in a neutral or friendly fashion. I went into a defensive posture, and tended to respond to questions with a lot of rambling detail rather than answering the question directly. I think this was a weakness in my presentation this time round. Jon's feedback at the end also also corroborated this.</p>
<p>I also made the mistake of not being thoroughly prepared for the Q&amp;A. The questions that were asked were questions that I had dealt with about 1.5 years ago, when I was in an earlier stage of the work I was presenting. However, because my focus now has shifted to other work, I was not prepared to tackle those questions again. I think at a bigger level, I didn't do the necessary "mental model" preparation for the talk, in which I should have rehearsed multiple "what-if" scenarios. (I've been reading Charles Duhigg's <a href="http://www.amazon.com/Smarter-Faster-Better-Productive-Business-ebook/dp/B00Z3FRYB0">Smarter Faster Better: The Secrets of Being Productive in Life and Business</a>, and the 'construction of mental models' is a core concept illustrated in there.)</p>
<p>To get around this, I think the next time I do a webinar, I should make the following preparations:</p>
<ul>
<li>Construct a mental model of the presentation as being a "dialogue", rather than a "defence". (I think it applies to all research presentations, really.)</li>
<li>A pen &amp; paper to jot down the question as it's being asked. It's a tool to slow my mind down, to prevent myself from making snap-judgment assumptions about what is being asked.</li>
<li>Be prepared to paraphrase the question back to the questioner prior to responding.</li>
<li>Prepare for a broad spectrum of questions to be answered, not just those that were recently asked in my context.</li>
<li>Recognize that it's totally okay to have problems picked out during a presentation, even one that is more of a “sales-pitch” type.</li>
</ul>
<p>There's always a first, and this first webinar was definitely a big educational experience for me. Though I think I didn't perform top-notch in the Q&amp;A portion, I hope I still left a positive impression on the IRD/ViPR team.</p>

  </div>

    <p><i>Did you enjoy this blog post? <a href="../../../blog/2016/3/22/webinar/#disqus">Let's discuss more</a>!</i></p>
    <hr class="fancy">
  
    
  <!-- Import stylesheet -->
  <link rel="stylesheet" href="../../../static/pygments.css">
  <div>

  <!-- Set title style -->
  
    <h1><a href="../../../blog/2016/3/21/paper-revisions/">Paper revisions</a></h1>
  

  <!-- Append author -->
  <p class="meta">
    written by
    
      <a href="https://twitter.com/ericmjl">Eric J. Ma</a>
    
    on 2016-03-21
  

  <!-- Put post body -->
  <p>We've submitted back our <a href="http://www.ericmajinglong.com/2015/12/03/reticulate-evolution-and-microbial-ecology-2/">reassortment paper</a> post-review. I've refrained from writing about this for about a week, mainly so that I can give myself enough emotional distance to reflect on it. On the whole, I'm feeling very thankful to the reviewers for their highly constructive reviews.</p>
<p>During the month-long review period, I was bracing myself emotionally for a thrashing review (esp. from a 3rd reviewer); after all I had seen such a fate befall two of my colleagues and a collaborator, all of whom wrote manuscripts on work that I was a part of. While there were constructive and legitimate criticisms in those reviewer reports, the number of comments that reflected more emotional ranting than reasoned argumentation left me baffled at the peer review process. (Aside: they mostly came from the fabled 3rd… and sometimes a 4th reviewer.)</p>
<p>When I first went over the reviews I was stunned reading the reviews; in my meeting with my advisor Jon about the reviews, he quipped that his "faith in the review process has been restored." Firstly, we didn't have to deal with the oft-fabled "3rd reviewer"; in the life sciences, the 3rd reviewer is often joked to be the one who shoots down a manuscript unreasonably. Secondly, the two reviewers gave us very encouraging and constructive comments. In retrospect, the reviewers comments were direct, thorough, and really helped us refine the scope of the claims we were making. (We may have overstepped the reasonable bounds of our claims in some places in the text.) Reviewer 1 was also very helpful in finding places where, because of familiarity with this manuscript, I had forgotten one chunk of text that should have been present. Reviewer 2 was overall very encouraging and supportive of the manuscript, and only raised points of clarification.</p>
<p>So, to the reviewers, thank you for the reviews. If I'm given the chance to do so, I will pass on to the next paper the same qualities of the reviews you gave me (directness, thoroughness, being reasoned). At this point, after all the editorial rejections by other journals, I'm just happy to have the paper reviewed, have its flaws pointed out. My hope is that our response was satisfactory, and I'm hoping we have a fair chance at having the reassortment paper published!</p>

  </div>

    <p><i>Did you enjoy this blog post? <a href="../../../blog/2016/3/21/paper-revisions/#disqus">Let's discuss more</a>!</i></p>
    <hr class="fancy">
  
    
  <!-- Import stylesheet -->
  <link rel="stylesheet" href="../../../static/pygments.css">
  <div>

  <!-- Set title style -->
  
    <h1><a href="../../../blog/2016/3/18/flask-jinja2-bootstrapcss-and-thoughts-on-being-a-maker/">Flask, Jinja2, bootstrap.css - and thoughts on being a maker</a></h1>
  

  <!-- Append author -->
  <p class="meta">
    written by
    
      <a href="https://twitter.com/ericmjl">Eric J. Ma</a>
    
    on 2016-03-18
  

  <!-- Put post body -->
  <p>I recently built a <a href="http://github.com/ericmjl/flu-gibson-webui/">front-end GUI</a> for one of my projects, which is a <a href="http://github.com/ericmjl/flu-gibson/">primer calculator</a> for doing Gibson assembly with influenza segments.</p>
<p>(<em>small detour</em>) The back-drop to this is that 7 years after the invention of the Gibson assembly method in the synthetic biology world (where I used to be), influenza researchers are still stuck with restriction cloning. If they switched to the Gibson (and other seamless) assembly methods, their time wasted on troubleshooting restriction cloning could be drastically reduced. I wrote this simple utility to help bridge that gap.</p>
<p>I had learned HTML many years back. I think I was still a teenager back then. So coming back to it now, I was both delighted and surprised to see the many new developments in the markup language.</p>
<p>Anyways, I built the FluGibson web interface with two intents. Firstly, for it to be a quick-and-dirty, simple utility that an experimental influenza researcher could use in their day-to-day. The user provides a nucleotide sequence and a name for that DNA part (something that’s convenient for the user to remember), and selects a standard plasmid backbone that the flu community uses. It returns a series of cloning primers and sequencing primers, and a PCR protocol for amplifying the DNA parts using the Phusion polymerase. Secondly, for it to be a prototype of a tool that I hope gets implemented in the <a href="../../../blog/2016/3/18/flask-jinja2-bootstrapcss-and-thoughts-on-being-a-maker/www.fludb.org/">Influenza Research Database</a>, where a researcher can do a one-click computation (and maybe even ordering) of the primers needed to clone an influenza segment from cDNA.</p>
<p>The FluGibson front-end is a Flask app, and as such runs in any modern browser. The backend is the FluGibson Python package that I wrote; it’s still a bit incomplete in terms of the examples, but given time, I’ll get those fixed up.</p>
<p>Being a maker, rather than a consumer, is a very empowering thing. Being a maker enables me to build what I need to have built to do what I need to get done. It takes time to learn the skill, but in the end I think it pays off. I’d like to encourage whoever’s reading this blog, go be a maker. Go make stuff that you think has tangible value for the world. It’s fun, it’s emotionally rewarding, and may bring a financial return. :-)</p>

  </div>

    <p><i>Did you enjoy this blog post? <a href="../../../blog/2016/3/18/flask-jinja2-bootstrapcss-and-thoughts-on-being-a-maker/#disqus">Let's discuss more</a>!</i></p>
    <hr class="fancy">
  

  
  <div class="pagination">
    
      <a href="../../../blog/page/33/">&laquo; Previous</a>
    
    | 34 |
    
      <a href="../../../blog/page/35/">Next &raquo;</a>
    
  </div>


    </div>

    <nav class="navbar navbar-expand-sm navbar-light fixed-bottom bg-light">
        <button class="navbar-toggler navbar-toggler-right" type="button" data-toggle="collapse" data-target="#external-links" aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
            <span class="navbar-toggler-icon"></span>
        </button>
        <div class="collapse navbar-collapse" id="external-links">
            <ul class="navbar-nav mr-auto">
                
                <li class="nav-item">
                    <a class="nav-link" href="https://www.linkedin.com/in/ericmjl"><i class="fab fa-linkedin" aria-hidden="true"></i> LinkedIn</a>
                </li>
                
                <li class="nav-item">
                    <a class="nav-link" href="http://twitter.com/ericmjl"><i class="fab fa-twitter" aria-hidden="true"></i> Twitter</a>
                </li>
                
                <li class="nav-item">
                    <a class="nav-link" href="http://github.com/ericmjl"><i class="fab fa-github" aria-hidden="true"></i> GitHub</a>
                </li>
                
                <li class="nav-item">
                    <a class="nav-link" href="http://stackoverflow.com/users/1274908/ericmjl"><i class="fab fa-stack-overflow" aria-hidden="true"></i> Stack Overflow</a>
                </li>
                
                <li class="nav-item">
                    <a class="nav-link" href="http://shortwhale.com/ericmjl"><i class="far fa-envelope" aria-hidden="true"></i> Contact Me</a>
                </li>
                
            </ul>
        </div>
    </nav>
    <!-- Boostrap JS imports -->
    <script src="https://code.jquery.com/jquery-3.3.1.slim.min.js" integrity="sha384-q8i/X+965DzO0rT7abK41JStQIAqVgRVzpbzo5smXKp4YfRvH+8abtTE1Pi6jizo" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.14.3/umd/popper.min.js" integrity="sha384-ZMP7rVo3mIykV+2+9J3UJ46jBk0WLaUAdn689aCwoqbBJiSnjAK/l8WvCWPIPm49" crossorigin="anonymous"></script>
    <script src="https://stackpath.bootstrapcdn.com/bootstrap/4.1.1/js/bootstrap.min.js" integrity="sha384-smHYKdLADwkXOn1EmN1qk/HfnUcbVRZyYmZ4qpPea6sjB/pTJ0euyQp0Mk8ck+5T" crossorigin="anonymous"></script>
</body>
